### PR TITLE
Improves code readability

### PR DESCRIPTION
**Pull request by kjivan**

### CODE REVIEW

File: src/array.ts [hunk 1]
```diff
@@ -1,31 +1,27 @@
-export const objectListForAttribute = <T>(
+export const groupByKey = <T>(
```
Comment (kjivan) [inline src/array.ts:1]: I think what you are doing is grouping  by key so I wanted to get a more concise name for it as opposed to the more abstract `objectListForAttribute`.

File: src/array.ts [hunk 2]
```diff
@@ -1,31 +1,27 @@
...
   objects: T[],
-  attribute: string
+  key: string
```
Comment (kjivan) [inline src/array.ts:3]: It's usually called an [object key](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys) so I thought that would be more descriptive.

File: src/array.ts [hunk 3]
```diff
@@ -1,31 +1,27 @@
...
-  return objects.reduce(
-    (objectListForAttribute: Record<string, T[]>, object: T) => {
-      objectListForAttribute[object[attribute]]
-        ? objectListForAttribute[object[attribute]].push(object)
-        : (objectListForAttribute[object[attribute]] = [object]);
-      return objectListForAttribute;
-    },
-    {}
-  );
+  return objects.reduce((grouped: Record<string, T[]>, object: T) => {
```
Comment (kjivan) [inline src/array.ts:5]: objectListForAttribute -> grouped I think improved the readability. Less abstract and easier to keep in your head I think.

File: src/array.ts [hunk 4]
```diff
@@ -1,31 +1,27 @@
...
+    const keyValue = object[key];
```
Comment (kjivan) [inline src/array.ts:6]: I typically try not to drill further than one value so I usually destructure or assign a const.

File: src/array.ts [hunk 5]
```diff
@@ -1,31 +1,27 @@
...
+    grouped[keyValue] = grouped[keyValue]
```
Comment (kjivan) [inline src/array.ts:7]: I tend to use functions that return values so I prefer the [spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) to the [Array.push](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) method. Similarly I prefer `slice` to `splice` I think it leads to more functional coding which is more readable and less buggy.

I don't like long method chains either. I usually try to break up method chains beyond 3-5 depending on the complexity to another function.

File: src/array.ts (hunk 1 above)
Comment (cravatsc) [reply src/array.ts:1]: makes sense, I always way overthink naming and it usually degrades after the first few minutes of thought

File: src/array.ts (hunk 5 above)
Comment (cravatsc) [reply src/array.ts:7]: Ok that is a good rule to go by. I also like the spread - its so natural for me to default to any collections built in methods from Java... on a similar vein, I was having a lot of trouble using `string[]` over `Array<string>`, it seems the syntactical sugar generally improves readability. Also like the rule of having a length to start evaluating a refactor

**Review by cravatsc — APPROVED** [review]

**Pull request merged**